roles/pve_exporter/tasks/main.yml

```
@@ -0,0 +1,87 @@
+---
+- name: Preflight
+  ansible.builtin.include_tasks:
+    file: preflight.yml
+  tags:
+    - pve_exporter_install
+    - pve_exporter_configure
+    - pve_exporter_run
+
+- name: Credentials
+  ansible.builtin.import_tasks: credentials.yml
+  when: pve_exporter_set_credential | bool
+
+- name: Install
+  tags:
+    - pve_exporter_install
+  block:
+    - name: Install packages dependencies
+      ansible.builtin.package:
+        name:
+          - python3
+          - python3-pip
+          - python3-virtualenv
+        state: present
```

You can make the python venv package install more dynamic like this:

```suggestion
    - name: Gather python facts
      ansible.builtin.setup:
        gather_subset:
          - "!all"
          - "python"
      when:
        "('python' not in ansible_facts) or
         ('discovered_interpreter_python' not in ansible_facts)

    - name: Install packages dependencies
      ansible.builtin.package:
        name:
          - "python{{ ansible_facts['ansible_python']['version']['major'] }}-setuptools"
          - "python{{ ansible_facts['ansible_python']['version']['major'] }}-venv"
        state: present
      become: true
```